components/contexts/Web3Context.js

```diff
@@ -46,6 +49,17 @@ export const Web3Provider = ({ children }) => {
     }
   }, [networkId])
 
+  const refreshToken = useCallback(async () => {
+    if (!contracts.LvlV1 || !address) return
+    const balance = Number(await contracts.LvlV1.balanceOf(address))
+    setHasLvlToken(balance > 0)
+  }, [contracts.LvlV1, address])
+
+  // Reload lvl token data whenever deps change:
+  useEffect(() => {
+    refreshToken()
+  }, [refreshToken, networkId, contracts.LvlV1, address])
```

Comment:
Is there any other thing we should be watching to detect a reload of the lvl token? This will catch:

* User switches network
* User switches account/address
* Contract changes (in case we ever need to do cross-network contracts or bridging)

![detect token](http://g.recordit.co/nssFCdN55Q.gif)